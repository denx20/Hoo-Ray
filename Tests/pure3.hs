{-# LANGUAGE LambdaCase #-}

add :: Int -> Int -> Int -> Int
add x y z = x + y + z

multiply :: Int -> Int -> Int
multiply x z = x * z

sub :: Int -> Int -> Int -> Int
sub k y = k - y

increment :: Int -> Int
increment x = x + 1

main :: IO ()
main = do
    let x = 10
    let y = 5
    let z = 3
    let w = add x y z
    let k = multiply x z
    let a = k `sub` y

    let b = increment a
    let c = increment b

    let d = add a b c
    let e = multiply d w
    let f = sub e y
    let g = add f x

    let h = increment g
    let i = increment h

    let j = add g h i
    let k' = multiply j f
    let l = sub k' h
    let m = add l g

    print m
