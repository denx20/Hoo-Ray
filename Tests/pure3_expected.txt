("w", ["add"])
("add", ["x","y","z"])
("k", ["multiply"])
("multiply", ["x","z"])
("a", ["sub"])
("sub", ["k","y"])
("b", ["increment"])
("increment", ["a"])
("c", ["increment"])
("increment", ["b"])
("d", ["add"])
("add", ["a","b","c"])
("e", ["multiply"])
("multiply", ["d","w"])
("f", ["sub"])
("sub", ["e","y"])
("g", ["add"])
("add", ["f","x"])
("h", ["increment"])
("increment", ["g"])
("i", ["increment"])
("increment", ["h"])
("j", ["add"])
("add", ["g","h","i"])
("k'", ["multiply"])
("multiply", ["j","f"])
("l", ["sub"])
("sub", ["k'","h"])
("m", ["add"])
("add", ["l","g"])
